homeassistant/components/blink/services.py

```diff
@@ -96,7 +88,9 @@ def collect_coordinators(
 
     async def async_handle_save_video_service(call: ServiceCall) -> None:
         """Handle save video service calls."""
-        camera_name = call.data[CONF_NAME]
+        registry = dr.async_get(hass)
+        device = registry.async_get(call.data[ATTR_DEVICE_ID][0])
+        camera_name = device.name if device else ""
```

Comment:
Why aren't the camera specific services entity services? Each camera entity corresponds to one camera api, right?

Reply:
That does make more sense.  I should have dug a bit deeper into how this was working (and breaking it).
I have moved the camera based services to the camera, left the pin service as is.
How should the PIN and UPDATE service be "connected" when they are API based - for each account that is logged in, we only need to call UPDATE and PIN one time for all devices/entities associated with the API instance, but I don't see a way to reference the API, only devices and entities.
For now, they work by picking a device or entity, but it doesn't feel like the best way to get there.

Reply:
Longterm I'd look into deprecating the refresh service. There doesn't seem to be a big difference between the coordinator refresh and the service. Both forces a refresh. The only difference seems to be that the refresh service doesn't change the last refresh time stored. Users can already use the `homeassistant.update_entity` service to automate refresh.

Reply:
To target a specific config entry we have the config entry selector which I'd use for a service that needs a specific coordinator only, like the send pin service.

Reply:
How do I go about depreciating this service - is there a message I should log or just update the docs?

Reply:
Create a repair issue

Reply:
I think I should do that in a separate PR right?

Reply:
Yes

Reply:
Reviewing this one more time... I'm not sure we should depreciate `blink.update`.  The update we "really want" is the config entry filtered version rather than by entity (which is provided in the HA.entity_update).  While picking an entity will refresh the whole API, I'm not sure it's clear that is what it is doing.

Reply:
Well. It's the same for all integrations so don't understand why we would do something different for this one❓